From 912114d3386ae33de3ddfc904b78c32dc5425565 Mon Sep 17 00:00:00 2001
From: Rich Megginson <rmeggins@redhat.com>
Date: Thu, 10 Jul 2014 14:18:06 -0600
Subject: [PATCH 243/243] Ticket #47692 single valued attribute replicated ADD
 does not work

https://fedorahosted.org/389/ticket/47692
Reviewed by: nhosoi (Thanks!)
Branch: 389-ds-base-1.2.11
Fix Description: Previous fix was incomplete - needed to move the
deleted attribute to the present attribute list.
Platforms tested: RHEL6 x86_64
Flag Day: no
Doc impact: no

(cherry picked from commit 4b54366593e50d71239588343f532eab39ef56e0)
(cherry picked from commit 60d8bf9f77c298197e4c8ecee585f8a2e6c0457c)
(cherry picked from commit 45dcda2425a865bf529ad0c8423664091392b865)
(cherry picked from commit aead2be2a9cc5392713c7849c18a9bf218beb533)
(cherry picked from commit 1dbf87ae7e9203ffe5968994701bfc8b71605c8f)
---
 ldap/servers/slapd/entrywsi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ldap/servers/slapd/entrywsi.c b/ldap/servers/slapd/entrywsi.c
index f184c7f..40fe39b 100644
--- a/ldap/servers/slapd/entrywsi.c
+++ b/ldap/servers/slapd/entrywsi.c
@@ -1184,6 +1184,8 @@ resolve_attribute_state_single_valued(Slapi_Entry *e, Slapi_Attr *a, int attribu
 		 */
 		/* just remove the deleted value */
 		entry_deleted_value_to_zapped_value(a,pending_value);
+		/* move the attribute to the present attributes list */
+		entry_deleted_attribute_to_present_attribute(e,a);
 		pending_value = NULL;
 		attr_set_deletion_csn(a,NULL);
 		return; /* we are done - we are keeping the present value */
-- 
1.8.1.4

